components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/GatewayArtifactsMgtDAO.java

```diff
@@ -216,29 +217,39 @@ private void updateGatewayPublishedAPIArtifacts(Connection connection, String ap
     }
 
     /**
-     * This method retrieves the Organization given the API UUID
+     * This method retrieves the organization and the deployed time of an API revision.
      *
-     * @param uuid API UUID
-     * @return Organization
-     * @throws APIManagementException If failed to retrieve organization
+     * @param apiUUID API UUID
+     * @param envName Environment name
+     * @param revisionUUId API revision UUID
+     * @return Organization and Deployed Time
+     * @throws APIManagementException If failed to retrieve organization and deployed time
      */
-    public String retrieveOrganization(String uuid) throws APIManagementException {
+    public OrgAndRevisionDeployedTimeInfoDTO retrieveOrgAndAPIRevisionDeployedTime(
+            String apiUUID, String envName, String revisionUUId) throws APIManagementException {
 
-        String query = SQLConstants.RETRIEVE_ORGANIZATION;
+        String query = SQLConstants.RETRIEVE_ORGANIZATION_AND_API_REVISION_DEPLOYED_TIME;
         String organization = null;
+        String deployedTime = null;
+        OrgAndRevisionDeployedTimeInfoDTO infoDTO = new OrgAndRevisionDeployedTimeInfoDTO();
         try (Connection connection = GatewayArtifactsMgtDBUtil.getArtifactSynchronizerConnection();
              PreparedStatement preparedStatement = connection.prepareStatement(query)) {
-            preparedStatement.setString(1, uuid);
+            preparedStatement.setString(1, apiUUID);
+            preparedStatement.setString(2, envName);
+            preparedStatement.setString(3, revisionUUId);
             try (ResultSet resultSet = preparedStatement.executeQuery()) {
                 while (resultSet.next()) {
                     organization = resultSet.getString("ORGANIZATION");
+                    deployedTime = resultSet.getString("DEPLOYED_TIME");
```

Comment:
Shall we use resultSet.getTimestamp method then no need to convert it to Date. We can use the long integer format date to the deployment.json file

Reply:
Fixed